addons/website_sale/static/src/js/website_sale.js

```diff
@@ -896,8 +921,30 @@ publicWidget.registry.productsSearchBar = publicWidget.Widget.extend({
                 widget: this,
             }));
             this.$menu.css('min-width', this.autocompleteMinWidth);
+
+            // Handle the case where the searchbar is in a mega menu by making
+            // it position:fixed and forcing its size. Note: this could be the
+            // default behavior or at least needed in more cases than the mega
+            // menu only (all scrolling parents). But as a stable fix, it was
+            // easier to fix that case only as a first step, especially since
+            // this cannot generically work on all scrolling parent.
+            const megaMenuEl = this.el.closest('.o_mega_menu');
+            if (megaMenuEl) {
+                const navbarEl = this.el.closest('.navbar');
+                const navbarTogglerEl = navbarEl ? navbarEl.querySelector('.navbar-toggler') : null;
+                if (navbarTogglerEl && navbarTogglerEl.clientWidth < 1) {
```

Comment:
I would have added a quick one line comment to mention this is to exclude "mobile" (or store it in a variable that would speak for itself) but that's fine

Reply:
Did it originally then found it explicit enough not too :shrug: